src/tests/unittests/hextobin.c

```diff
@@ -0,0 +1,123 @@
+/*
+ * hextobin.c: Test suite for sc_hex_to_bin()
+ *
+ * Copyright (C) 2022 Peter Popovec <popovec.peter@gmail.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdio.h>
+#include <stdint.h>
+#include <string.h>
+#include "libopensc/opensc.h"
+
+#define LEN 15
+
+int main()
+{
+	char *t;
+	char test[][LEN] = {
+/* result length (0x60 error string, 0x70 end of array), string, result string */
```

Comment:
I would suggest to use some struct instead of two-dimensional array with comment to describe its functionality. Something like this would make both the structure and the code more readable: 
```
struct {
	int result_len;
	const char *input;
	const char *output;
} test[] = {
	1, "0", "\x00",
	1, ":0", "\x00",
	...
};
```
(or even a bit more verbose with separating results and expected output length with proper types)